tests/functional/base_test.go

```diff
@@ -63,22 +74,125 @@ func DefaultDataplaneNodeSetTemplate(name types.NamespacedName, spec dataplanev1
 	}
 }
 
+func DefaultNodeSpec() dataplanev1.OpenStackDataPlaneNodeSetSpec {
```

Comment:
I suggest to switch to unstructured create so that i) you don't have to define unfilled fields ii) you can simulate if the client does not send a field in the json request.

[1] https://github.com/openstack-k8s-operators/lib-common/blob/19e1d7506ae86e1f391f0bfbcac31e228507d999/modules/common/test/helpers/common.go#L87-L111 
[2] https://github.com/openstack-k8s-operators/nova-operator/commit/8aaaaee7d8ef68dd49172b786f117bb5e0aa50fb

Reply:
Technically I could leave those fields unfilled. Decorators in type definition are taking care of these. Mostly I've left them like this to guide future test development. If defaults or the way controller manages them it will be more obvious what to adjust. Same goes for tightening tests.

But I understand it's a question of taste.

Reply:
While my first point can be a matter of taste. The second cannot. If you use golang structs as input in your test then you cannot simulate that the user left a field out from the json request. Take the Deploy flag as an example. It is defined as 
```golang
	// +kubebuilder:default=true
	Deploy bool `json:"deploy"`
```
If you want to test that it is properly defaulted to "true" then you need to send in an input json request where the deploy field is unset. You cannot do that with a golang struct.

Reply:
Ack. I'll add rewrite of existing tests to use unstructured to this PR.